sky/backends/cloud_vm_ray_backend.py

```diff
@@ -4084,6 +4085,70 @@ def _execute_storage_mounts(self, handle: CloudVmRayResourceHandle,
         end = time.time()
         logger.debug(f'Storage mount sync took {end - start} seconds.')
 
+    def _execute_storage_csync(
+            self, handle: CloudVmRayResourceHandle,
+            storage_mounts: Dict[Path, storage_lib.Storage]) -> None:
+        """Executes continuous syncing on storages.
+
+        This function only runs the CSYNC daemon on the given storage
+        and the files/dirs to be copied to remote node are handled in
+        the _execute_file_mounts.
+
+        """
+        storage_mounts = {
+            path: storage_mount
+            for path, storage_mount in storage_mounts.items()
+            if storage_mount.mode == storage_lib.StorageMode.C_SYNC
+        }
+
+        if not storage_mounts:
+            return
+
+        cloud = handle.launched_resources.cloud
+        if isinstance(cloud, clouds.Local):
+            logger.warning(
+                f'{colorama.Fore.YELLOW}Sky On-prem does not support '
+                'storage syncing. No action will be taken.'
+                f'{colorama.Style.RESET_ALL}')
+            return
+
+        fore = colorama.Fore
+        style = colorama.Style
+        plural = 's' if len(storage_mounts) > 1 else ''
+        logger.info(f'{fore.CYAN}Processing {len(storage_mounts)} '
+                    f'storage continuous sync{plural}.{style.RESET_ALL}')
+        start = time.time()
+        ip_list = handle.external_ips()
+        assert ip_list is not None, 'external_ips is not cached in handle'
+        ssh_credentials = backend_utils.ssh_credential_from_yaml(
+            handle.cluster_yaml)
+        runners = command_runner.SSHCommandRunner.make_runner_list(
+            ip_list, **ssh_credentials)
+        log_path = os.path.join(self.log_dir, 'storage_csyncs.log')
+
+        for src, storage_obj in storage_mounts.items():
+            if not os.path.isabs(src) and not src.startswith('~/'):
+                src = f'{SKY_REMOTE_WORKDIR}/{src}'
+            # Get the first store and use it to sync
+            store = list(storage_obj.stores.values())[0]
+            csync_cmd = store.csync_command(src, storage_obj.interval)
+            src_print = (storage_obj.source
+                         if storage_obj.source else storage_obj.name)
+            if isinstance(src_print, list):
+                src_print = ', '.join(src_print)
+            backend_utils.parallel_data_transfer_to_nodes(
+                runners,
+                source=src_print,
+                target=src,
+                cmd=csync_cmd,
+                run_rsync=False,
+                action_message='Setting cont. sync',
+                log_path=log_path,
+            )
```

Comment:
Should we handle the exception for this function call just as what we did in the mounted mode `_execute_storage_mounts`?

Reply:
Thanks for catching this. The `dst` directory will already be occupied with files as it copies over the files in the `src` directory opposed to the `MOUNT` mode. Hence, I implemented the general error handling only for `_execute_storage_csync` without `exceptions.MOUNT_PATH_NON_EMPTY_CODE`.